db.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"math"
+	"os"
+	"strings"
+	"time"
+	"unicode"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+type saveDb struct {
+	db        *sql.DB
+	tableName string
+}
+
+const (
+	dbLocation              = "./tcping.db"
+	eventTypeStatistics     = "statistics"
+	eventTypeHostnameChange = "hostname change"
+	tableSchema             = `
+CREATE TABLE %s (
+    id INTEGER PRIMARY KEY,
+    event_type TEXT NOT NULL, -- for the data type eg. statistics, hostname change
+    timestamp DATETIME,
+    addr TEXT,
+    hostname TEXT,
+    port INTEGER,
+    hostname_resolve_tries INTEGER,
+
+    hostname_changed_to TEXT,
+    hostname_change_time DATETIME,
+
+    latency_min REAL,
+    latency_avg REAL,
+    latency_max REAL,
+
+	total_duration TEXT,
+    start_timestamp DATETIME,
+    end_timestamp DATETIME,
+
+    last_successful_probe DATETIME,
+    last_unsuccessful_probe DATETIME,
+
+    longest_uptime TEXT,
+    longest_uptime_end DATETIME,
+    longest_uptime_start DATETIME,
+
+    longest_downtime TEXT,
+    longest_downtime_start DATETIME,
+    longest_downtime_end DATETIME,
+
+    total_packets INTEGER,
+    total_packet_loss REAL,
+    total_successful_probes INTEGER,
+    total_unsuccessful_probes INTEGER,
+
+    total_uptime TEXT,
+    total_downtime TEXT
+);`
+)
+
+func newDb(args []string, dbPath string) saveDb {
```

Comment:
Please make sure all the functions/methods are having their documentation.

The correct format is the name of the function/method followed by its use case.

For instance https://github.com/pouriyajamshidi/tcping/blob/6a418489e474884f049f67d9d58e7cb88a0abc55/tcping.go#L143